src/ziggurat/timestamp_transformer.clj

```diff
@@ -22,10 +26,14 @@
          (^void init [_ ^ProcessorContext context]
            (set! processor-context context))
          (transform [_ record-key record-value]
-           ;;(log/debug "stream record metadata--> " "record-key: " record-key " record-value: " record-value " partition: " (.partition processor-context) " topic: " (.topic processor-context))
-           (let [message-time (.timestamp processor-context)]
-             (when (message-to-process? message-time oldest-processed-message-in-s)
-               (calculate-and-report-kafka-delay metric-namespace message-time additional-tags)
+           ;; (log/debug "stream record metadata--> " "record-key: " record-key " record-value: " record-value " partition: " (.partition processor-context) " topic: " (.topic processor-context))
+           (let [timestamp (.timestamp processor-context)
```

Comment:
Why not create a different function to extract metadata? Doing this inside the timestamp transformer doesn't make sense.

Reply:
It is already a different function to extract the metadata...